mender-image-prepare.sh

```diff
@@ -0,0 +1,535 @@
+#!/bin/bash
+
+show_help() {
+cat << EOF
+Usage: $0 [prepare_image | make_sdimg | make_sdcard] [-i|--image | -d|--device -r|--rootsize]
+
+        device - can be /dev/sd* or /dev/mmcblk*
+
+        prepare_image - shrinks existing embedded image
+                        obligatory parameters: --image
+
+        make_sdcard   - formats, partitions and sets the sd card
+                        obligatory parameters: --image & --rootsize
+                        
+                        note: rootsize can found as an output from
+                        'prepare_image' command
+
+        make_sdimg    - composes image file compliant with Mender partition
+                        table scheme
+
+EOF
+}
+
+if [ $# -eq 0 ]; then
+  show_help
+  exit 1
+fi
+
+if ! id | grep -q root; then
+        echo "must be run as root"
+        exit
+fi
+
+BASE_DIR=${PWD}
+
+# Erase block size 8MiB
+erase_block=8388608
+# Boot partition start in sectors (512 bytes per sector).
+pboot_start=16384
+# Boot partition size in sectors.
+pboot_size=$(( 16777216 / 512 ))
+# Data partition size in sectors.
+pdata_size=$(( 134217728 / 512 ))
+
+# beaglebone: 4521984
+prootfs_size=
+target_dev=
+image=
+number_of_parts=
+loopsdimg=
+
+declare -a part_names=("boot" "primary" "secondary" "data")
```

Comment:
I believe that we have refereed to the rootfs parts as "rootfsA" and "rootfsB" in other locations, especially in Yocto. Might be nice to be consistent but this is a minor.

Reply:
These names are taken from demo image for BeagleBone Black.
With command `lsblk -o name,mountpoint,label,size,uuid` you can easily verify it.